shared/4-hydrogen-vm-render-script.js

```diff
@@ -147,12 +149,46 @@ async function mountHydrogen() {
   // page don't say `undefined`.
   urlRouter.attach();
 
+  const storageFactory = new StorageFactory(
+    null,
+    window.indexedDB,
+    window.FDBKeyRange,
+    window.localStorage
+  );
+  const storage = await new Promise((resolve) => {
+    platform.logger.run('creating-storage', async (log) => {
+      const storage = await storageFactory.create('1', log);
+      resolve(storage);
+    });
+  });
+
+  const hsApi = new HomeServerApi({
+    request: (/*url, options*/) => {
+      return {
+        abort() {},
+        async response() {
+          // Respond to the `/context` requests about missing event relations
+          // with empty content. We don't want more errors to be logged but
+          // we're ok with the missing events staying missing.
+          return {
+            // 204: No content
+            status: 204,
+            body: {},
+          };
+        },
+      };
+    },
+    homeserver: 'fakehomeserver',
+  });
+
   // We use the timeline to setup the relations between entries
   const timeline = new Timeline({
     roomId: roomData.id,
     fragmentIdComparer: fragmentIdComparer,
     clock: platform.clock,
     logger: platform.logger,
+    storage,
+    hsApi,
```

Comment:
We could get away with removing these again but since we [might have to use these two interfaces (http and storage)](https://github.com/MadLittleMods/hydrogen-static-archives-prototype/issues/1) in the future, we might as well get a little familiar.

Reply:
Removed in https://github.com/matrix-org/matrix-public-archive/pull/43/commits/74b096c1fcdd9ba7889d49d96ea12bb4b9a5c0a0 because not necessary for this fix and it was causing problems with the CI, https://github.com/matrix-org/matrix-public-archive/pull/43#discussion_r927231258